drivers/at86rf2xx/aes_spi.c

```diff
@@ -0,0 +1,356 @@
+    _aes_transfer_bytes(dev, true, &cfg, NULL, sizeof(cfg));
+    _aes_transfer_bytes(dev, true, src, NULL, 1);
+    _aes_transfer_bytes(dev, true, src + 1, dst, AT86RF2XX_AES_BLOCK_SIZE - 1);
+    _aes_transfer_bytes(dev, false, &mirror,
+                        dst ? dst + AT86RF2XX_AES_BLOCK_SIZE - 1 : NULL, 1);
+}
+
+void at86rf2xx_aes_key_read_encrypt(at86rf2xx_t *dev,
+                                    uint8_t key[AT86RF2XX_AES_KEY_LENGTH])
+{
+    uint8_t cfg = AT86RF2XX_AES_CTRL_AES_MODE__KEY |
+                  AT86RF2XX_AES_CTRL_AES_DIR__ENC;
+    at86rf2xx_spi_get_bus(dev);
+    _aes_open_write(dev, AT86RF2XX_REG__AES_CTRL);
+    _aes_transfer_bytes(dev, false, &cfg, NULL, sizeof(cfg));
+    _aes_open_read(dev, AT86RF2XX_REG__AES_KEY_START);
+    _aes_transfer_bytes(dev, false, NULL, key, AT86RF2XX_AES_KEY_LENGTH);
+    at86rf2xx_spi_release_bus(dev);
+}
+
+void at86rf2xx_aes_key_write_encrypt(at86rf2xx_t *dev,
+                                     const uint8_t key[AT86RF2XX_AES_KEY_LENGTH])
+{
+    uint8_t cfg = AT86RF2XX_AES_CTRL_AES_MODE__KEY |
+                  AT86RF2XX_AES_CTRL_AES_DIR__ENC;
+    at86rf2xx_spi_get_bus(dev);
+    _aes_open_write(dev, AT86RF2XX_REG__AES_CTRL);
+    _aes_transfer_bytes(dev, true, &cfg, NULL, sizeof(cfg));
+    _aes_transfer_bytes(dev, false, key, NULL, AT86RF2XX_AES_KEY_LENGTH);
+    at86rf2xx_spi_release_bus(dev);
+}
+
+void at86rf2xx_aes_key_read_decrypt(at86rf2xx_t *dev,
+                                    uint8_t key[AT86RF2XX_AES_KEY_LENGTH])
+{
+    uint8_t cfg = AT86RF2XX_AES_CTRL_AES_MODE__KEY |
+                  AT86RF2XX_AES_CTRL_AES_DIR__DEC;
+    at86rf2xx_spi_get_bus(dev);
+    _aes_open_write(dev, AT86RF2XX_REG__AES_CTRL);
+    _aes_transfer_bytes(dev, false, &cfg, NULL, sizeof(cfg));
+    _aes_open_read(dev, AT86RF2XX_REG__AES_KEY_START);
+    _aes_transfer_bytes(dev, false, NULL, key, AT86RF2XX_AES_KEY_LENGTH);
+    at86rf2xx_spi_release_bus(dev);
+}
+
+void at86rf2xx_aes_key_write_decrypt(at86rf2xx_t *dev,
+                                     const uint8_t key[AT86RF2XX_AES_KEY_LENGTH])
+{
+    uint8_t cfg = AT86RF2XX_AES_CTRL_AES_MODE__KEY |
+                  AT86RF2XX_AES_CTRL_AES_DIR__DEC;
+    at86rf2xx_spi_get_bus(dev);
+    _aes_open_write(dev, AT86RF2XX_REG__AES_CTRL);
+    _aes_transfer_bytes(dev, true, &cfg, NULL, sizeof(cfg));
+    _aes_transfer_bytes(dev, false, key, NULL, AT86RF2XX_AES_KEY_LENGTH);
+    at86rf2xx_spi_release_bus(dev);
+}
+
+int at86rf2xx_aes_ecb_encrypt(at86rf2xx_t *dev,
+                              aes_block_t *cipher,
+                              uint8_t key[AT86RF2XX_AES_BLOCK_SIZE],
+                              const aes_block_t *plain,
+                              uint8_t nblocks)
+{
+    if (!nblocks) {
+        return 0;
+    }
+    uint8_t cfg = AT86RF2XX_AES_CTRL_AES_MODE__ECB |
+                  AT86RF2XX_AES_CTRL_AES_DIR__ENC;
+    uint8_t mirror = cfg | AT86RF2XX_AES_CTRL_MIRROR_MASK__AES_REQUEST;
```

Comment:
What's that mirror about?

Reply:
The `mirror` must be the same value as the `cfg` and is transfered to the slave to start an AES operation, if `AT86RF2XX_AES_CTRL_MIRROR_MASK__AES_REQUEST` is set.